packages/ag-charts-community/src/scale/ordinalTimeScale.ts

```diff
@@ -113,12 +137,22 @@ export class OrdinalTimeScale extends BandScale<Date, TimeInterval | number> {
 
     private findInterval(target: number) {
         // Binary search for the target
-        const { sortedTimestamps } = this;
-        let low = 0;
-        let high = sortedTimestamps.length - 1;
+        const { sortedTimestamps, precomputedSteps } = this;
+        let low: number;
+        let high: number;
+        if (precomputedSteps != null) {
+            const d0 = sortedTimestamps[0].valueOf();
+            const d1 = sortedTimestamps[sortedTimestamps.length - 1].valueOf();
+            const i = (((target - d0) / (d1 - d0)) * precomputedSteps.length) | 0;
+            low = precomputedSteps[i];
+            high = i < precomputedSteps.length - 1 ? precomputedSteps[i + 1] : sortedTimestamps.length - 1;
+        } else {
+            low = 0;
+            high = sortedTimestamps.length - 1;
+        }
 
         while (low <= high) {
-            const mid = Math.floor((low + high) / 2);
+            const mid = ((low + high) / 2) | 0;
```

Comment:
Is `| 0` correct?

Reply:
It's a trick to floor which is more efficient.
I'd leave a comment here, because not all people know these sort of hacks.

Reply:
It's a performance hack for JS engines - it means they know this is an int32. It's equivalent to Math.trunc - rounds towards zero